### PR TITLE
feat: add `autoscaler_policy_overrides` support to module

### PR DESCRIPTION
**Pull request by aweris**
Bumps `castai` provider `v7.2.0` and add new field `autoscaler_policy_overrides` and deprecation notice to `autoscaler_policies_json`

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -657,8 +657,140 @@ resource "helm_release" "castai_kvisor_self_managed" {
...
+          enabled                             = try(spot_instances.value.enabled, null)
+          max_reclaim_rate                    = try(spot_instances.value.max_reclaim_rate, null)
+          spot_diversity_enabled              = try(spot_instances.value.spot_diversity_enabled, null)
+          spot_diversity_price_increase_limit = try(spot_instances.value.spot_diversity_price_increase_limit, null)
+
+          dynamic "spot_backups" {
+            for_each = try(spot_instances.value.spot_backups, null) != null ? [spot_instances.value.spot_backups] : []
+
+            content {
+              enabled                          = try(spot_backups.value.enabled, null)
```
Comment (linkas45) [inline main.tf:745]: if spot_backups is null, still all good?

File: main.tf [hunk 2]
```diff
@@ -657,8 +657,140 @@ resource "helm_release" "castai_kvisor_self_managed" {
...
+            for_each = try(node_downscaler.value.empty_nodes, null) != null ? [node_downscaler.value.empty_nodes] : []
+
+            content {
+              enabled       = try(empty_nodes.value.enabled, null)
+              delay_seconds = try(empty_nodes.value.delay_seconds, null)
+            }
+          }
+
+          dynamic "evictor" {
+            for_each = try(node_downscaler.value.evictor, null) != null ? [node_downscaler.value.evictor] : []
```
Comment (linkas45) [inline main.tf:777]: why we cannot have this ` for_each = try(node_downscaler.value.evictor, [])`?

Comment (aweris) [reply main.tf:777]: because of the `[node_downscaler.value.evictor]` transformation in then statement

Comment (aweris) [reply main.tf:777]: but double check if TF is handing this magically. If it can, going to replace it

Comment (aweris) [reply main.tf:777]: ` for_each = try([autoscaler_policy_overrides.value.unschedulable_pods], [])` this works

**Review by linkas45 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -91,6 +91,53 @@ module "castai_gke_cluster" {
...
   }
+
+  autoscaler_policy_overrides = {
+    enabled                                 = true
+    node_templates_partial_matching_enabled = false
+
+    unschedulable_pods = {
+      enabled = true
+
+      headroom = {
```
Comment (linkas45) [inline README.md:102]: what is headroom? I don't see it in original policies json

Comment (aweris) [reply README.md:102]: it's part of the `unschedulable pods` in the policy object. 

https://gitlab.com/castai/kubecast/-/blame/master/api-contracts/grpc/proto/policies/v1/policy.proto?ref_type=heads#L60

**Review by aldor007 — APPROVED** [review]

**Pull request merged**